### DISCUSSION TITLE
How can I use the VuePress markdown settings and plugins in a custom component?

### DISCUSSION THREAD

**Discussion by royston-c-oa** [vuepress/core#641] [Q&A] [answered]
I was wondering if there is a way to use the `@vuepress/markdown` package in a custom component.

My use can is that I have an OpenAPI spec I'd like to convert into a document. This means the the OpenAPi spec is in a .json file that I load into the component then render out response code samples to the UI using `openapi-sampler`.

I was hoping to use the same settings I have setup with VuePress (`lineNumbers` etc) in the custom component.

When I try to load `import { createMarkdown } from "@vuepress/markdown";` I get the error shown below. I suspect because it's a node module?

```
Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'dirname')
```

Any help would be greatly appreciated. 😃

**Comment by Mister-Hope** [comment, marked as answer]
Yes, the package are using node apis, and is designed running in node env, if you want somthing like this running in browsers, you may need to fork a version and make some "edits" yourself.
